lib/Sema/CSDiagnostics.cpp

```diff
@@ -4732,7 +4732,8 @@ bool MissingGenericArgumentsFailure::diagnoseParameter(
   if (locator->isForContextualType() &&
       llvm::count_if(cs.DefaultedConstraints,
                      [&GP](const ConstraintLocator *locator) {
-                       return locator->getGenericParameter() == GP;
+                       return (locator->isForGenericParameter() &&
```

Comment:
Is it necessary to check for `isForGenericParameter` first because `getGenericParameter` would return `nullptr` in case it's not which wouldn't be equal to a given `GP`?

Reply:
Here's the implementation of `ConstraintLocator::getGenericParameter`:
```
GenericTypeParamType *ConstraintLocator::getGenericParameter() const {
  return castLastElementTo<LocatorPathElt::GenericParameter>().getType();
}
```
So, if the last element is not `LocatorPathElt::GenericParameter`, this will crash. Of course, I could change the implementation to behave like `TypeVariableType::Implementation::getGenericParameter`, which does do the check first. What do you think?

Reply:
I didn't realize that it got re-implemented that way! :) I think it would make sense to check inside and return `nullptr` if type variable is not associated with generic parameter...

Reply:
I'm pretty sure there is still a bunch of code that replies on the old behavior too...